antarest/study/storage/variantstudy/model/command/create_link.py

```diff
@@ -57,11 +53,128 @@ def validate_series(
+        }
+        for attr in MATRIX_ATTRIBUTES:
+            if value := getattr(self, attr, None):
+                args[attr] = strip_matrix_protocol(value)
+        return CommandDTO(action=self.command_name.value, args=args, study_version=self.study_version)
+
+    def match(self, other: ICommand, equal: bool = False) -> bool:
+        if not isinstance(other, CreateLink):
+            return False
+        simple_match = self.area1 == other.area1 and self.area2 == other.area2
+        if not equal:
+            return simple_match
+        return (
+            simple_match
+            and self.parameters == other.parameters
+            and self.series == other.series
+            and self.direct == other.direct
+            and self.indirect == other.indirect
+        )
+
+    def _create_diff(self, other: "ICommand") -> List["ICommand"]:
+        other = cast(CreateLink, other)
+
+        commands: List[ICommand] = []
+        area_from, area_to = sorted([self.area1, self.area2])
+        if self.parameters != other.parameters:
+            properties = LinkInternal.model_validate(other.parameters or {}).model_dump(
+                mode="json", by_alias=True, exclude_none=True, exclude={"area1", "area2"}
+            )
+            commands.append(
+                UpdateConfig(
+                    target=f"input/links/{area_from}/properties/{area_to}",
+                    data=properties,
+                    command_context=self.command_context,
+                    study_version=self.study_version,
+                )
+            )
+        if self.series != other.series:
+            commands.append(
+                ReplaceMatrix(
+                    target=f"@links_series/{area_from}/{area_to}",
+                    matrix=strip_matrix_protocol(other.series),
+                    command_context=self.command_context,
+                    study_version=self.study_version,
+                )
+            )
+        return commands
+
+    def get_inner_matrices(self) -> List[str]:
+        list_matrices = []
+        for attr in MATRIX_ATTRIBUTES:
+            if value := getattr(self, attr, None):
+                assert_this(isinstance(value, str))
+                list_matrices.append(strip_matrix_protocol(value))
+        return list_matrices
+
+    def save_series(self, area_from: str, area_to: str, study_data: FileStudy, version: StudyVersion) -> None:
+        self.series = self.series or (self.command_context.generator_matrix_constants.get_link(version=version))
+        assert isinstance(self.series, str)
+        if version < STUDY_VERSION_8_2:
+            study_data.tree.save(self.series, ["input", "links", area_from, area_to])
+        else:
+            study_data.tree.save(
+                self.series,
+                ["input", "links", area_from, f"{area_to}_parameters"],
+            )
+            study_data.tree.save({}, ["input", "links", area_from, "capacities"])
```

Comment:
Not really sure why you do this, is it to ensure we don't have both matrices writing ? I think you should remove this. 
Same goes for the 2 save_ functions

Reply:
I'm not sure why. It was there before